src/ui/components/userDetail/userRoles/UserRolesList.tsx

```diff
@@ -12,41 +12,73 @@
  * License for the specific language governing permissions and limitations
  * under the License.
  */
-import { useState } from "react";
+import { useContext, useEffect, useState } from "react";
 import { ReactComponent as CrossIcon } from "../../../../assets/cross.svg";
 import { ReactComponent as PlusSquareIcon } from "../../../../assets/plus-square.svg";
 import Badge from "../../badge";
 import { LayoutPanel } from "../../layout/layoutPanel";
 import { UserRolesListHeader } from "./UserRolesListHeader";
 
+import { useUserRolesService } from "../../../../api/userroles/user/roles";
+import { getImageUrl } from "../../../../utils";
+import { PopupContentContext } from "../../../contexts/PopupContentContext";
 import Alert from "../../alert";
 import Select from "../../select";
+import Shimmer from "../../shimmer";
 import AssignRolesDialog from "../../userroles/components/dialogs/AssignRoles";
 import DeleteUserRoleDialog from "../../userroles/components/dialogs/DeleteUserRole";
 import { useUserDetailContext } from "../context/UserDetailContext";
 import "./userRolesList.scss";
 
 type UserRolesListProps = {
-	currentlySelectedTenantId: string;
 	userId: string;
-	roles: string[];
-	isFeatureEnabled: boolean;
-	onTenantIdChange: (tenantId: string) => void;
 };
 
-export default function UserRolesList(props: UserRolesListProps) {
-	const { roles, isFeatureEnabled, userId } = props;
+export default function UserRolesList({ userId }: UserRolesListProps) {
+	const { getRolesForUser } = useUserRolesService();
+	const { userDetail } = useUserDetailContext();
+	const { showToast } = useContext(PopupContentContext);
 
-	const [assignedRoles, setAssignedRoles] = useState<string[]>(roles);
+	const [assignedRoles, setAssignedRoles] = useState<string[] | undefined>(undefined);
+	const [isFeatureEnabled, setIsFeatureEnabled] = useState<boolean | undefined>(undefined);
 
 	const [showAddRoleDialog, setShowAddRoleDialog] = useState(false);
 	const [roleToDelete, setRoleToDelete] = useState<undefined | string>(undefined);
-	const [showDeleteRoleDialog, setShowDeleteDialogRole] = useState(true);
+	const [showDeleteRoleDialog, setShowDeleteDialogRole] = useState(false);
+
+	const tenantIdsThatUserIsPartOf = userDetail.details.tenantIds;
+	const [currentlySelectedTenantId, setCurrentlySelectedTenantId] = useState(tenantIdsThatUserIsPartOf[0]);
```

Comment:
if they are a part of the public tenant, we show that, else we show the 0th index.